sdk/core/azure_core/src/task/mod.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! Asynchronous task execution utilities.
+
+use std::{fmt, future::Future, pin::Pin, sync::Arc};
+
+#[cfg(all(not(feature = "tokio"), not(target_arch = "wasm32")))]
+mod standard_spawn;
+
+#[cfg(all(not(feature = "tokio"), target_arch = "wasm32"))]
+mod wasm_spawn;
+
+#[cfg(feature = "tokio")]
+mod tokio_spawn;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(all(not(feature = "tokio"), not(target_arch = "wasm32")))]
+pub use standard_spawn::StdSpawner;
+
+#[cfg(all(not(feature = "tokio"), target_arch = "wasm32"))]
+pub use wasm_spawn::WasmSpawner;
+
+#[cfg(feature = "tokio")]
+pub use tokio_spawn::TokioSpawner;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(crate) type TaskFuture = Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
+
+#[cfg(target_arch = "wasm32")]
+pub(crate) type TaskFuture = Pin<Box<dyn Future<Output = ()> + 'static>>;
```

Comment:
Why not just use `async_trait` like we do everywhere else? I don't recall if it's `Pin`, but it'll rewrite to a `BoxedFuture` and consistent with all other code. Should work here, too.

Reply:
`async_trait` doesn't map futures as parameters to trait functions :(. So unfortunately I need to make it explicit.